app/models/concerns/pre_sale_prices/line_item_pre_saleable.rb

```diff
@@ -0,0 +1,20 @@
+module PreSalePrices
+  module LineItemPreSaleable
+    extend ActiveSupport::Concern
+
+    included do
+      before_validation :copy_pre_sale_price
```

Comment:
Possibly, yeah. I was trying to match the behaviour of the price setting: https://github.com/spree/spree/blob/master/core/app/models/spree/line_item.rb#L16